Frontend/ui-driver/src/Flow.purs

```diff
@@ -1866,26 +1869,29 @@ onBoardingSubscriptionScreenFlow onBoardingSubscriptionViewCount = do
-  _ <- pure $ printLog "Registration token" (getValueToLocalStore REGISTERATION_TOKEN)
-  _ <- pure $ cleverTapSetLocation unit
-  if (getValueToLocalNativeStore IS_RIDE_ACTIVE) == "true" && (not $ any (\item -> isLocalStageOn item) [RideAccepted, RideStarted, ChatWithCustomer]) then currentRideFlow Nothing
-    else pure unit
-  (GlobalState globalState) <- getState
-  getDriverInfoResp <- getDriverInfoDataFromCache (GlobalState globalState) false
-  when globalState.homeScreen.data.config.subscriptionConfig.enableBlocking $ do checkDriverBlockingStatus getDriverInfoResp
-  when globalState.homeScreen.data.config.subscriptionConfig.completePaymentPopup $ checkDriverPaymentStatus getDriverInfoResp
-  updateBannerAndPopupFlags
-  liftFlowBT hideSplash
-  void $ lift $ lift $ toggleLoader false
-  when (globalState.globalProps.addTimestamp && globalState.homeScreen.props.tobeLogged) $ do
-    liftFlowBT $ setEventTimestamp "homeScreenFlow"
-    logData <- liftFlowBT $ getTimeStampObject unit
-    liftFlowBT $ logEventWithMultipleParams logField_ "sending_logs" logData
-    modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen { props {tobeLogged = false}})
-    modifyScreenState $ GlobalPropsType $ \globalProps -> globalProps{addTimestamp = false}
-  liftFlowBT $ handleUpdatedTerms $ getString TERMS_AND_CONDITIONS_UPDATED
+  Events.measureDurationFlowBT "Flow.homeScreenFlow" $ do    
+    _ <- pure $ delay $ Milliseconds 1.0
+    _ <- pure $ printLog "Registration token" (getValueToLocalStore REGISTERATION_TOKEN)
+    _ <- pure $ cleverTapSetLocation unit
+    if (getValueToLocalNativeStore IS_RIDE_ACTIVE) == "true" && (not $ any (\item -> isLocalStageOn item) [RideAccepted, RideStarted, ChatWithCustomer]) then currentRideFlow Nothing
+      else pure unit
+    (GlobalState globalState) <- getState
+    getDriverInfoResp <- getDriverInfoDataFromCache (GlobalState globalState) false
+    when globalState.homeScreen.data.config.subscriptionConfig.enableBlocking $ do checkDriverBlockingStatus getDriverInfoResp
+    when globalState.homeScreen.data.config.subscriptionConfig.completePaymentPopup $ checkDriverPaymentStatus getDriverInfoResp
+    updateBannerAndPopupFlags
+    liftFlowBT hideSplash
+    void $ lift $ lift $ toggleLoader false
+    when (globalState.globalProps.addTimestamp && globalState.homeScreen.props.tobeLogged) $ do
+      liftFlowBT $ setEventTimestamp "homeScreenFlow"
+      logData <- liftFlowBT $ getTimeStampObject unit
+      liftFlowBT $ logEventWithMultipleParams logField_ "sending_logs" logData
+      modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen { props {tobeLogged = false}})
+      modifyScreenState $ GlobalPropsType $ \globalProps -> globalProps{addTimestamp = false}
+    liftFlowBT $ handleUpdatedTerms $ getString TERMS_AND_CONDITIONS_UPDATED
+  liftFlowBT $ Events.initMeasuringDuration "Screen.homeScreen"
   action <- UI.homeScreen
+  _ <- lift $ lift $ Remote.pushSDKEvents
```

Comment:
This will always make API calls whenever home screen flow is exited and looped again. The data in backend will be appended with this or replaced? for same keys. Also this API we need to make async, otherwise it can take some time to process the homeScreen output action

Reply:
Yes, resolved made it forked. Also, I will only push the first time an action comes from Homescreen, then onwards I will prevent as for now our use case is only to track App Load time so we need not call the API many times for a same sessionId.